dataset/bin.cpp

```diff
@@ -453,9 +453,32 @@ DataArray groupby_concat_bins(const DataArrayConstView &array,
       builder.groups(), {reductionDim});
 }
 
+namespace {
+void validate_bin_args(const std::vector<VariableConstView> &edges,
+                       const std::vector<VariableConstView> &groups) {
+  if (edges.empty()) {
+    if (groups.empty()) {
+      throw except::BucketError(
+          "Arguments 'edges' and 'groups' of scipp.bin are "
+          "both empty. At least one must be set.");
+    }
+  } else {
+    for (std::size_t i = 0; i < edges.size(); ++i) {
```

Comment:
Can you write this as `for (const auto &var : edges)` and then below output the actual `dim` that is responsible instead of just the index `i`?